include/ncengine/graphics/SkeletalAnimator.h

```diff
@@ -13,38 +13,96 @@ namespace nc::graphics
+
         /**
          * @brief Construct a new SkeletalAnimator component.
          * @param entity The parent entity.
          * @param meshUid The UID of the mesh that will be animated by this component.
          * @param animationUid The UID of the asset::SkeletalAnimation that will become the initial animation for this component. The initial state will always loop.
          */
-        SkeletalAnimator(Entity entity, std::string meshUid, std::string animationUid);
+        SkeletalAnimator(Entity entity, const std::string& meshUid, const std::string& animationUid);
         SkeletalAnimator(SkeletalAnimator&& other) noexcept  = default;
         SkeletalAnimator(SkeletalAnimator& other) = delete;
         auto operator=(SkeletalAnimator&& other) noexcept -> SkeletalAnimator&  = default;
         auto operator=(SkeletalAnimator& other) -> SkeletalAnimator& = delete;
         ~SkeletalAnimator() noexcept = default;
 
-        /** @brief Called by SkeletalAnimationSystem. Executes the state machine; all animations whose transition conditions are met emit a signal to be processed. */
+        /** 
+         * @brief Called by SkeletalAnimationSystem. Executes the state machine.
+         * All animations whose transition conditions are met emit a signal to be processed.
+         * */
         void Execute();
 
+        /** 
+         * @brief Add an anim::Loop state to the state machine.
+         * @param loopProperties Properties required to create an anim::Loop state.
+         */
+        auto AddState(const anim::Loop& loopProperties) -> uint32_t;
+
+        /** 
+         * @brief Add an anim::PlayOnce state to the state machine.
+         * @param playOnceProperties Properties required to create an anim::PlayOnce state.
+         */
+        auto AddState(const anim::PlayOnce& playOnceProperties) -> uint32_t;
+
+        /** 
+         * @brief Add an anim::Stop state to the state machine.
+         * @param stopProperties Properties required to create an anim::Stop state.
+         */
+        auto AddState(const anim::Stop& stopProperties) -> uint32_t;
+
+        /**
+         * @brief Interrupt the state machine with an immediate looping animation.
+         * Transitions back into the state machine when exitWhen is met.
+         * @param animUid The animation to play.
+         * @param exitWhen Exit back to the state machine when this condition is met.
+         * @param exitTo Exit back to this state.
+         */
+        void LoopImmediate(const std::string& animUid, const std::function<bool ()>& exitWhen, uint32_t exitTo);
+
+        /**
+         * @brief Interrupt the state machine with an immediate animation that plays once.
+         * Transitions back into the state machine when the animation has completed.
+         * @param animUid The animation to play
+         * @param exitTo Exit back to this state.
+         */
+        void PlayOnceImmediate(const std::string& animUid, uint32_t exitTo);
```

Comment:
Not sure how much sense this makes, but I think it would be conventient if the `exitTo` params had a default. Is it sensible to just go back to the root/initial state, if nothing else is specified?